### DISCUSSION TITLE
Latest version crashing for long simulations with SPM model

### DISCUSSION THREAD

**Discussion by ganmad** [pybamm-team/PyBaMM#3444] [Q&A]
Hello,

It was mentioned on the release note that latest version was optimised for speed and memory usage. However, I noticed the opposite with my simulations.  I noticed that ageing simulations have become slower and crashing for longer cycles, even for SPM model.  I tried different methods to speed up the solver, but it is still slower for SPM. Below code has crashed after 41% of progress. Can someone check whether the case is the same with you as well or it is only the problem with my installation please? Here is the minimal example. Let me know if I missed anything. I use WSL (Ubuntu 22) on Windows and updated recently with latest PyBaMM version. Python 3.10.12

```
import pybamm
model = pybamm.lithium_ion.SPM({"thermal": "lumped","SEI": "solvent-diffusion limited","SEI film resistance":"distributed",
        "SEI porosity change": "true",})
param = pybamm.ParameterValues("OKane2022")

safe_solver = pybamm.CasadiSolver(atol=1e-3, rtol=1e-3, mode="safe",dt_max=2000)
exp = pybamm.Experiment(["Rest for 8 hours (120 minute period)", "Discharge at 0.5C until 2.5V (15 minute period)","Charge at 0.5C until 4.1V (15 minute period)",]*7300,termination="70% capacity")
sim = pybamm.Simulation(model,parameter_values=param,experiment=exp, solver=safe_solver)
sol = sim.solve(showprogress=True,save_at_cycles=50)
```

Here is the error that was shown after crash on Jupyter notebook on VS code: 
`The Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. View Jupyter [log](command:jupyter.viewOutput) for further details`

which could be due to eating up of RAM by PyBaMM.

Thanks.

**Comment by ganmad** [comment]
@brosaplanella @Saransh-cpp Can you please check whether latest version works fine in your PC for longer simulations, using above minimal code? Thanks.

**Comment by kratman** [comment]
Was your crash from running out of memory? I was trying out the block of code you shared, and the estimate from the process monitor was 16GB of memory by around 30% of the simulation.

**Reply by ganmad** [reply]
That's true. I see an extensive usage of the memory towards the crashing of the simulation. For example, during one of the runs before crash: 
![image](https://github.com/pybamm-team/PyBaMM/assets/50149439/8a26840c-ba02-4589-9fff-9b07fed4058d)

I ran such simulations in previous version installation without a problem. Is this particularly the case with the new version? But how did you estimate the memory? That's quite interesting to know how to estimate.

**Reply by kratman** [reply]
I was just using the mac "Activity Monitor" app. I ran the current pybamm (23.9rc0) using pycharm to run the notebook and just a regular jupyter notebook to run with version 23.5. I got similar memory behavior on both. Which version were you using before?

**Reply by ganmad** [reply]
Thanks for checking on your system. I used 22.9 previously.

**Reply by kratman** [reply]
Ok. If I have time I will try that version to see what happens, but that is a year of pybamm updates between 22.9 and the current version

**Reply by brosaplanella** [reply]
It could be related to the new Experiment set up which processes them more efficiently in terms of memory.

**Reply by ganmad** [reply]
@kratman Great, that will be helpful to understand where the issue is. Thanks for looking into this.

**Reply by ganmad** [reply]
> It could be related to the new Experiment set up which processes them more efficiently in terms of memory.

May be
